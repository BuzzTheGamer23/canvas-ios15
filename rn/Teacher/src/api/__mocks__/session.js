/* @flow */

const template = require('../canvas-api/__templates__/session')

export function getSession (): ?Session {
  return template.session()
}
