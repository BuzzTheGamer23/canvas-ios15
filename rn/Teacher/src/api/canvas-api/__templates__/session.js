/* @flow */

import template from '../../utils/template'

export const session: Template<Session> = template({
  authToken: 'iamanauthtoken',
  baseURL: 'http://mobiledev.instructure.com',
  user: {
    name: 'Key and Peele',
    avatar_url: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
    primary_email: 'keyandpeele@instructure.com',
  },
})
