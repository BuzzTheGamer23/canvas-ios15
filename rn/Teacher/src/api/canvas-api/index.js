// @flow

import * as coursesApi from './courses'
import * as usersApi from './users'
import * as enrollmentsApi from './enrollments'
import * as submissionsApi from './submissions'
import * as assignmentsApi from './assignments'
import * as quizzesApi from './quizzes'
import * as assignmentGroupsApi from './assignmentGroups'
import * as discussionsApi from './discussions'
import * as conversationApi from './conversations'

type CombinedApi = $Supertype<typeof coursesApi
  & typeof usersApi
  & typeof enrollmentsApi
  & typeof submissionsApi
  & typeof assignmentsApi
  & typeof assignmentGroupsApi
  & typeof quizzesApi
  & typeof discussionsApi
  & typeof conversationApi>

export default ({
  ...coursesApi,
  ...usersApi,
  ...enrollmentsApi,
  ...submissionsApi,
  ...assignmentsApi,
  ...quizzesApi,
  ...assignmentGroupsApi,
  ...discussionsApi,
  ...conversationApi,
}: CombinedApi)
