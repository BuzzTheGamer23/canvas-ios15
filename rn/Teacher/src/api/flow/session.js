// @flow

export type Session = {
  authToken: string,
  baseURL: string,
}
