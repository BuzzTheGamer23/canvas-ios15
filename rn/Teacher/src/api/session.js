/* @flow */

export type Session = {
  authToken: string,
  baseURL: string,
}

let currentSession: ?Session = null

export function setSession (session: ?Session) {
  currentSession = session
}

export function getSession (): ?Session {
  return currentSession
}
