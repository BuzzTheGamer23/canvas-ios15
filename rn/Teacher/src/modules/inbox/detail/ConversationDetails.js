// @flow

import React, { Component } from 'react'
import {
  View,
  FlatList,
  StyleSheet,
  TouchableOpacity,
  Image,
  ActionSheetIOS,
} from 'react-native'
import { connect } from 'react-redux'
import refresh from '../../../utils/refresh'
import { default as InboxActions } from '../actions'
import Screen from '../../../routing/Screen'
import branding from '../../../common/branding'
import i18n from 'format-message'
import ConversationMessage from '../components/ConversationMessageRow'
import { Heading1 } from '../../../common/text'
import color from '../../../common/colors'
import Images from '../../../images'

const {
  refreshConversationDetails,
  starConversation,
  unstarConversation,
  deleteConversation,
} = InboxActions

const Actions = {
  refreshConversationDetails,
  starConversation,
  unstarConversation,
  deleteConversation,
}

export type ConversationOwnProps = {
  conversation: ?Conversation,
  conversationID: string,
  messages: ConversationMessage[],
  pending?: boolean,
  error?: ?string,
}

export type RefreshProps = {
  refresh?: Function,
  refreshing?: boolean,
  refreshConversationDetails: Function,
  starConversation: Function,
  unstarConversation: Function,
}

export type ConversationDetailsProps = ConversationOwnProps & RefreshProps & NavigationProps

export class ConversationDetails extends Component <any, ConversationDetailsProps, any> {
  constructor (props: ConversationDetailsProps) {
    super(props)

    this.state = {
      deletePending: false,
    }
  }

  _renderItem = ({ item, index }) => {
    return <ConversationMessage
              navigator={this.props.navigator}
              message={item}
              conversation={this.props.conversation}
              onReplyButtonPressed={ () => {} }
              firstMessage={index === 0} />
  }

  _toggleStarred = () => {
    if (!this.props.conversation) return
    if (this.props.conversation.starred) {
      this.props.unstarConversation(this.props.conversationID)
    } else {
      this.props.starConversation(this.props.conversationID)
    }
  }

  _renderHeader = () => {
    if (!this.props.conversation) return <View />

    const star = this.props.conversation.starred ? Images.starFilled : Images.starLined

    return (<View style={styles.header}>
              <Heading1>{this.props.conversation.subject || i18n('(no subject)')}</Heading1>
              <TouchableOpacity onPress={this._toggleStarred} hitSlop={{ top: 8, bottom: 8, left: 8, right: 8 }}>
                <View style={{ backgroundColor: 'white' }}>
                  <Image source={star} style={{ tintColor: branding.primaryBrandColor, height: 24, width: 24 }}/>
                </View>
              </TouchableOpacity>
            </View>)
  }

  _renderComponent = () => {
    const header = this._renderHeader()
    return (
      <View style={styles.container}>
        <FlatList
          style={styles.list}
          data={this.props.messages}
          renderItem={this._renderItem}
          ListHeaderComponent={header}
          refreshing={this.props.refreshing}
          onRefresh={this.props.refresh}
        />
      </View>
    )
  }

  render () {
    return (
      <Screen
        navBarColor={branding.navBarColor}
        navBarStyle='dark'
        drawUnderNavBar={true}
        title={i18n('Message Details')}
        rightBarButtons={[
          {
            image: Images.kabob,
            testID: 'inbox.detail.options.button',
            action: this.showOptionsActionSheet,
          },
        ]}
      >
        { this._renderComponent() }
      </Screen>
    )
  }

  componentWillReceiveProps (nextProps: ConversationDetailsProps) {
    if (this.state.deletePending && !nextProps.pending && !nextProps.conversation) {
      this.setState({ deletePending: false })
      this.props.navigator.pop()
    }
  }

  showOptionsActionSheet = () => {
    const options = [
      i18n('Delete'),
      i18n('Cancel'),
    ]
    ActionSheetIOS.showActionSheetWithOptions(
      {
        options,
        destructiveButtonIndex: options.length - 2,
        cancelButtonIndex: options.length - 1,
      },
      this.handleOptionsActionSheet,
    )
  }

  handleOptionsActionSheet = (index: number) => {
    switch (index) {
      case 0:
        this.deleteConversation()
        break
    }
  }

  deleteConversation () {
    this.setState({ deletePending: true })
    // $FlowFixMe
    this.props.deleteConversation(this.props.conversationID)
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F5F5F5',
  },
  loading: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  list: {
    backgroundColor: '#F5F5F5',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: 'white',
    padding: global.style.defaultPadding,
    borderTopWidth: StyleSheet.hairlineWidth,
    borderTopColor: color.seperatorColor,
  },
})

export function mapStateToProps (state: AppState, props: any): ConversationOwnProps {
  const inbox = state.inbox
  const conversationID = props.conversationID
  const convoState = inbox.conversations[conversationID]
  let messages = []
  let conversation: ?Conversation
  if (convoState &&
    convoState.data) {
    conversation = convoState.data
    messages = convoState.data.messages || []
  }

  return {
    conversation,
    conversationID,
    messages,
    pending: convoState ? convoState.pending > 0 : false,
    error: convoState ? convoState.error : null,
  }
}

export function handleRefresh (props: ConversationDetailsProps): void {
  props.refreshConversationDetails(props.conversationID)
}

export function shouldRefresh (props: ConversationOwnProps): boolean {
  if (!props.conversation) return true
  return !props.conversation.messages
}

export const Refreshed: any = refresh(
  handleRefresh,
  shouldRefresh,
  props => Boolean(props.pending)
)(ConversationDetails)
const Connected = connect(mapStateToProps, Actions)(Refreshed)
export default (Connected: Component<any, ConversationDetailsProps, any>)
