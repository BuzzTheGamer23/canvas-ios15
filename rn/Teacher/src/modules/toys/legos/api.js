/* this obviously won't be in the same module when we actually do API stuff */

export function buyLegoSet (legoSet: LegoSet): Promise<LegoSet> {
  return new Promise((resolve, reject) => {
    let action = () => {
      if (Math.random() < 0.3) {
        reject()
      } else {
        resolve(legoSet)
      }
    }
    setTimeout(action, 1000)
  })
}

export function sellAllLegos (): Promise<void> {
  return new Promise((resolve) => setTimeout(resolve, 1000))
}
