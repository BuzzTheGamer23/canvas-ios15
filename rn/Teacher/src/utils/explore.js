/* @flow */

function hasProp (object: any, key: string, value: any): boolean {
  if (object instanceof Object) {
    const props = object.props
    if (props && props instanceof Object) {
      for (var prop in props) {
        if (prop === key && props[prop] === value) {
          return true
        }
      }
    }
  }

  return false
}

function selectByProp (object: any, propKey: string, value: any): ?any {
  if (!(typeof object === 'object')) {
    return null
  }

  if (hasProp(object, propKey, value)) {
    return object
  }

  let result = null

  for (var key in object) {
    if (object[key] instanceof Array) {
      object[key].forEach((e) => {
        result = selectByProp(e, propKey, value)
        if (result) {
          return result
        }
      })
    }

    result = selectByProp(object[key], propKey, value)
    if (result) {
      return result
    }
  }

  return result
}

class ComponentExplorer {
  component: any

  constructor (component: any) {
    this.component = component
  }

  selectByProp (propKey: string, value: any): ?any {
    return selectByProp(this.component, propKey, value)
  }

  selectByID (id: string): ?any {
    return this.selectByProp('testID', id)
  }
}

export default function explore (component: any): ComponentExplorer {
  return new ComponentExplorer(component)
}
