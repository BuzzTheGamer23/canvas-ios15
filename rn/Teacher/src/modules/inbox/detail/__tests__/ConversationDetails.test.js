/* @flow */
import {
  ActionSheetIOS,
} from 'react-native'
import React from 'react'
import { ConversationDetails, mapStateToProps, handleRefresh, shouldRefresh, type ConversationDetailsProps } from '../ConversationDetails.js'
import { setSession } from '../../../../api/session'
import explore from '../../../../../test/helpers/explore'
import setProps from '../../../../../test/helpers/setProps'

const template = {
  ...require('../../../../api/canvas-api/__templates__/conversations'),
  ...require('../../../../api/canvas-api/__templates__/users'),
  ...require('../../../../__templates__/helm'),
  ...require('../../../../redux/__templates__/app-state'),
  ...require('../../../../api/canvas-api/__templates__/session'),
}

jest
  .mock('TouchableHighlight', () => 'TouchableHighlight')
  .mock('../../../../routing/Screen')

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

const Screen = (props: ConversationDetailsProps) => {
  const screen = renderer.create(<ConversationDetails {...props} />)
  const tree = () => screen.toJSON()
  return {
    component: screen,
    get optionsButton (): any {
      return explore(tree()).selectRightBarButton('inbox.detail.options.button')
    },

    tapOptionsButton () {
      this.optionsButton.action()
      return this
    },

    testRender () {
      expect(tree()).toMatchSnapshot()
    },
  }
}

describe('ConversationDetails', () => {
  let props
  beforeEach(() => {
    props = {
      conversation: template.conversation({ id: '1' }),
      conversationID: '1',
      messages: [],
      refreshConversationDetails: jest.fn(),
      starConversation: jest.fn(),
      unstarConversation: jest.fn(),
      deleteConversation: jest.fn(),
      navigator: template.navigator(),
    }
  })

  beforeAll(() => {
    setSession(template.session())
  })

  it('renders correctly', () => {
    Screen(props).testRender()
  })

  it('renders correctly with some messages', () => {
    // $FlowFixMe
    props.messages = [template.conversationMessage()]
    Screen(props).testRender()
  })

  it('shows options action sheet', () => {
    // $FlowFixMe
    ActionSheetIOS.showActionSheetWithOptions = jest.fn()
    Screen(props).tapOptionsButton()
    expect(ActionSheetIOS.showActionSheetWithOptions).toHaveBeenCalledWith(
      {
        options: ['Delete', 'Cancel'],
        destructiveButtonIndex: 0,
        cancelButtonIndex: 1,
      },
      expect.any(Function),
    )
  })

  it('calls deleteConversation from options', () => {
    // $FlowFixMe
    ActionSheetIOS.showActionSheetWithOptions = jest.fn((config, callback) => callback(config.options.length - 2))
    props.deleteConversation = jest.fn()
    props.conversationID = '1'
    Screen(props).tapOptionsButton()
    expect(props.deleteConversation).toHaveBeenCalledWith('1')
  })

  it('calls pop after delete finishes', () => {
    // $FlowFixMe
    ActionSheetIOS.showActionSheetWithOptions = jest.fn((config, callback) => callback(config.options.length - 2))
    props.navigator.pop = jest.fn()
    const screen = Screen(props)
    props.deleteConversation = jest.fn(() => {
      setProps(screen.component, { pending: 1 })
      setProps(screen.component, { pending: 0, conversation: null })
    })
    screen.component.update(<ConversationDetails {...props} />)
    screen.tapOptionsButton()
    expect(props.navigator.pop).toHaveBeenCalled()
  })
})

it('mapStateToProps', () => {
  const c1 = template.conversation({
    id: '1',
  })
  const appState = template.appState({
    inbox: {
      conversations: {
        '1': { data: c1 },
      },
      selectedScope: 'all',
      all: { refs: [c1.id] },
      unread: { refs: [] },
      starred: { refs: [] },
    },
  })

  const results = mapStateToProps(appState, { conversationID: c1.id })
  expect(results).toMatchObject({
    conversation: c1,
  })
})

it('handleRefresh', () => {
  const props = {
    refreshConversationDetails: jest.fn(),
    starConversation: jest.fn(),
    unstarConversation: jest.fn(),
    conversationID: '1',
    conversation: null,
    messages: [],
    navigator: template.navigator(),
  }

  handleRefresh(props)
  expect(props.refreshConversationDetails).toHaveBeenCalledWith('1')
})

it('should refresh', () => {
  const conversation = template.conversation({ messages: null })
  const props = {
    conversation,
    conversationID: conversation.id,
    messages: [],
  }

  expect(shouldRefresh(props)).toEqual(true)
})
