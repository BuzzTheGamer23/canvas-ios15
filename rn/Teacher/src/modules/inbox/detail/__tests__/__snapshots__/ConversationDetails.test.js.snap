// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationDetails renders correctly 1`] = `
<Screen
  drawUnderNavBar={true}
  navBarColor="#374A59"
  navBarStyle="dark"
  rightBarButtons={
    Array [
      Object {
        "action": [Function],
        "image": 1,
        "testID": "inbox.detail.options.button",
      },
    ]
  }
  title="Message Details"
>
  <View
    style={
      Object {
        "backgroundColor": "#F5F5F5",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      ListHeaderComponent={
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderTopColor": "#C7CDD1",
              "borderTopWidth": 0.5,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 16,
            }
          }
        >
          <Heading1>
            learn to write code
          </Heading1>
          <TouchableOpacity
            accessibilityLabel="Unstarred"
            accessibilityTraits="button"
            activeOpacity={0.2}
            focusedOpacity={0.7}
            hitSlop={
              Object {
                "bottom": 8,
                "left": 8,
                "right": 8,
                "top": 8,
              }
            }
            onPress={[Function]}
            testID="inbox.detail.not-starred"
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 24,
                    "tintColor": "#374A59",
                    "width": 24,
                  }
                }
              />
            </View>
          </TouchableOpacity>
        </View>
      }
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={undefined}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onViewableItemsChanged={undefined}
      refreshing={undefined}
      renderItem={[Function]}
      renderScrollComponent={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "backgroundColor": "#F5F5F5",
        }
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderTopColor": "#C7CDD1",
                "borderTopWidth": 0.5,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 16,
              }
            }
          >
            <Text
              accessibilityTraits="header"
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontFamily": ".SFUIDisplay-semibold",
                    "fontSize": 20,
                  },
                  undefined,
                ]
              }
            >
              learn to write code
            </Text>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel="Unstarred"
              accessibilityTraits="button"
              accessible={true}
              hitSlop={
                Object {
                  "bottom": 8,
                  "left": 8,
                  "right": 8,
                  "top": 8,
                }
              }
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="inbox.detail.not-starred"
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 24,
                      "tintColor": "#374A59",
                      "width": 24,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</Screen>
`;

exports[`ConversationDetails renders correctly with some messages 1`] = `
<Screen
  drawUnderNavBar={true}
  navBarColor="#374A59"
  navBarStyle="dark"
  rightBarButtons={
    Array [
      Object {
        "action": [Function],
        "image": 1,
        "testID": "inbox.detail.options.button",
      },
    ]
  }
  title="Message Details"
>
  <View
    style={
      Object {
        "backgroundColor": "#F5F5F5",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      ListHeaderComponent={
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderTopColor": "#C7CDD1",
              "borderTopWidth": 0.5,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 16,
            }
          }
        >
          <Heading1>
            learn to write code
          </Heading1>
          <TouchableOpacity
            accessibilityLabel="Unstarred"
            accessibilityTraits="button"
            activeOpacity={0.2}
            focusedOpacity={0.7}
            hitSlop={
              Object {
                "bottom": 8,
                "left": 8,
                "right": 8,
                "top": 8,
              }
            }
            onPress={[Function]}
            testID="inbox.detail.not-starred"
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 24,
                    "tintColor": "#374A59",
                    "width": 24,
                  }
                }
              />
            </View>
          </TouchableOpacity>
        </View>
      }
      data={
        Array [
          Object {
            "attachments": Array [
              Object {
                "display_name": "something.pdf",
                "id": "1",
                "mime_class": "pdf",
              },
            ],
            "author_id": "1234",
            "body": "this is neat",
            "id": "123456789",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={undefined}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onViewableItemsChanged={undefined}
      refreshing={undefined}
      renderItem={[Function]}
      renderScrollComponent={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "backgroundColor": "#F5F5F5",
        }
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderTopColor": "#C7CDD1",
                "borderTopWidth": 0.5,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 16,
              }
            }
          >
            <Text
              accessibilityTraits="header"
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontFamily": ".SFUIDisplay-semibold",
                    "fontSize": 20,
                  },
                  undefined,
                ]
              }
            >
              learn to write code
            </Text>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel="Unstarred"
              accessibilityTraits="button"
              accessible={true}
              hitSlop={
                Object {
                  "bottom": 8,
                  "left": 8,
                  "right": 8,
                  "top": 8,
                }
              }
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="inbox.detail.not-starred"
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 24,
                      "tintColor": "#374A59",
                      "width": 24,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
        >
          <View
            testID="inbox.conversation-message-123456789"
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderTopColor": "#C7CDD1",
                  "borderTopWidth": 0.5,
                  "flex": 1,
                  "padding": 16,
                  "paddingTop": 12,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "#C7CDD1",
                    "borderBottomWidth": 0.5,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingBottom": 12,
                  }
                }
              >
                <View
                  accessibilityLabel="artorias of the abyss + 1 others to me "
                  accessible={true}
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "height": 32,
                        "marginRight": 8,
                        "width": 32,
                      }
                    }
                  >
                    <View
                      accessibilityLabel=""
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "white",
                            "borderColor": "#C7CDD1",
                            "borderWidth": 0.5,
                            "justifyContent": "center",
                            "overflow": "hidden",
                          },
                          Object {
                            "borderRadius": 16,
                            "height": 32,
                            "width": 32,
                          },
                        ]
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        disabled={false}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "fontWeight": "600",
                            },
                            Object {
                              "fontSize": 11,
                            },
                          ]
                        }
                      >
                        AOT
                      </Text>
                    </View>
                  </View>
                  <View>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "fontFamily": ".SFUIDisplay",
                          },
                          Object {
                            "color": "#2D3B45",
                            "fontSize": 16,
                          },
                          undefined,
                          Object {
                            "fontFamily": ".SFUIDisplay",
                          },
                        ]
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        disabled={false}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "fontFamily": ".SFUIDisplay",
                            },
                            Object {
                              "color": "#2D3B45",
                              "fontSize": 16,
                            },
                            Object {
                              "color": "#2D3B45",
                              "fontSize": 14,
                              "fontWeight": "600",
                            },
                            Object {
                              "fontFamily": ".SFUIDisplay-semibold",
                            },
                          ]
                        }
                      >
                        artorias of the abyss + 1 others 
                      </Text>
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        disabled={false}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "fontFamily": ".SFUIDisplay",
                            },
                            Object {
                              "color": "#2D3B45",
                              "fontSize": 16,
                            },
                            Object {
                              "color": "#8B969E",
                              "fontSize": 14,
                            },
                            Object {
                              "fontFamily": ".SFUIDisplay",
                            },
                          ]
                        }
                      >
                        to me
                      </Text>
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "fontFamily": ".SFUIDisplay",
                          },
                          Object {
                            "color": "#2D3B45",
                            "fontSize": 16,
                          },
                          Object {
                            "color": "#8B969E",
                            "fontSize": 12,
                          },
                          Object {
                            "fontFamily": ".SFUIDisplay",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel="Message options"
                  accessibilityTraits="button"
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "flex-end",
                      "height": 32,
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 32,
                    }
                  }
                  testID="conversation-message.kabob-123456789"
                  tvParallaxProperties={undefined}
                >
                  <Image
                    source={1}
                    style={
                      Object {
                        "height": 18,
                        "tintColor": "#8B969E",
                        "transform": Array [
                          Object {
                            "rotate": "180deg",
                          },
                        ],
                        "width": 18,
                      }
                    }
                  />
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "paddingTop": 16,
                  }
                }
                testID={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  numberOfLines={0}
                  style={
                    Array [
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                    ]
                  }
                >
                  this is neat
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="inbox.conversation-message-123456789.attachment-1"
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "marginTop": 8,
                    }
                  }
                >
                  <Image
                    source={1}
                    style={
                      Object {
                        "tintColor": "#008EE2",
                      }
                    }
                  />
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontFamily": ".SFUIDisplay",
                        },
                        Object {
                          "color": "#2D3B45",
                          "fontSize": 16,
                        },
                        Object {
                          "color": "#008EE2",
                          "fontFamily": ".SFUIDisplay-bold",
                          "fontSize": 14,
                          "marginLeft": 6,
                        },
                        Object {
                          "fontFamily": ".SFUIDisplay",
                        },
                      ]
                    }
                  >
                    something.pdf
                  </Text>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits="button"
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "marginTop": 8,
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontFamily": ".SFUIDisplay",
                        },
                        Object {
                          "color": "#2D3B45",
                          "fontSize": 16,
                        },
                        Array [
                          Object {
                            "color": "#374A59",
                            "fontSize": 14,
                          },
                          Object {
                            "fontFamily": ".SFUIDisplay-bold",
                          },
                          Object {
                            "color": "#374A59",
                          },
                          Object {},
                          Object {
                            "fontSize": 16,
                            "fontWeight": "500",
                          },
                        ],
                        Object {
                          "fontFamily": ".SFUIDisplay-medium",
                        },
                      ]
                    }
                  >
                    Reply
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</Screen>
`;
